cwlgen/workflowdeps.py

```diff
@@ -0,0 +1,253 @@
+
+        """
+        Parameter.__init__(self, param_id=param_id, label=label,
+                           secondary_files=secondary_files, param_format=param_format,
+                           streamable=streamable, doc=doc, param_type=param_type, requires_type=False)
+        self.inputBinding = input_binding
+        self.default = default
+
+    @classmethod
+    def parse_with_id(cls, d, identifier):
+        if isinstance(d, str):
+            d = {"type": d}
+        d["id"] = identifier
+        return super(InputParameter, cls).parse_dict(d)
+
+
+class WorkflowStepInput(Serializable):
+    """
+    The input of a workflow step connects an upstream parameter (from the workflow inputs, or the outputs of
+    other workflows steps) with the input parameters of the underlying step.
+
+    Documentation: https://www.commonwl.org/v1.0/Workflow.html#WorkflowStepInput
+    """
+    def __init__(self, input_id, source=None, link_merge=None, default=None, value_from=None):
+        """
+        :param input_id: A unique identifier for this workflow input parameter.
+        :type input_id: STRING
+        :param source: Specifies one or more workflow parameters that will provide input to the underlying step parameter.
+        :type source: STRING | list[STRING]
+        :param link_merge: The method to use to merge multiple inbound links into a single array.
+                           If not specified, the default method is "merge_nested".
+        :type link_merge: LinkMergeMethod
+        :param default: The default value for this parameter to use if either there is no source field,
+                        or the value produced by the source is null
+        :type default: Any | DictRepresentible
+        :param value_from: If valueFrom is a constant string value, use this as the value for this input parameter.
+                           If valueFrom is a parameter reference or expression,
+                           it must be evaluated to yield the actual value to be assiged to the input field.
+        :type value_from: STRING
+        """
+
+        self.id = input_id
+        self.source = source
+        self.linkMerge = parse_link_merge_method(link_merge)
+        self.default = default
+        self.valueFrom = value_from
+
+
+    @classmethod
+    def parse_with_id(cls, d, identifier):
+        if isinstance(d, str):
+            d = {"source": d}
+        d["id"] = identifier
+        return super(WorkflowStepInput, cls).parse_dict(d)
+
+
+class WorkflowStep(Serializable):
+    """
+    A workflow step is an executable element of a workflow. It specifies the underlying process implementation
+    (such as CommandLineTool or another Workflow) in the run field and connects the input and output parameters
+    of the underlying process to workflow parameters.
+
+    Documentation: https://www.commonwl.org/v1.0/Workflow.html#WorkflowStep
+    """
+
+    parse_types = [("inputs", [[WorkflowStepInput]])]
```

Comment:
Why do you not parse outputs?

Reply:
Eagle eye ;)

Reply:
I'm confused though, I looked further down and your parsing test seemed to find the outputs? Why did this not catch it: https://github.com/common-workflow-language/python-cwlgen/pull/22/files#diff-519c8f8c384a01398241be92a9878e72R100